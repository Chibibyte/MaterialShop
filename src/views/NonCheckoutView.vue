<template>
  <div>
    <Navbar />
    <v-main>
      <router-view />
    </v-main>
  </div>
</template>


<script>
/**
 * Business Logic
 * -display Navbar
 * -contain router-view for NonCheckoutViews
 *
 * Inputs: none
 * Outputs: none
 */
import Navbar from "@/components/Navbar.vue";
export default {
  name: "NonCheckoutView",
  components: {
    Navbar
  }
};
</script>