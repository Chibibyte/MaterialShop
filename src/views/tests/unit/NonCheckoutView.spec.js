
import Vuex from 'vuex';
import { shallowMount, createLocalVue } from '@vue/test-utils';
import NonCheckoutView from '@/views/NonCheckoutView.vue';

const localVue = createLocalVue();
localVue.use(Vuex);

let wrapper;

beforeEach(() => {
    wrapper = shallowMount(NonCheckoutView, {
        stubs: [
            'v-main',
            'router-view'
        ],
        localVue
    })
})

describe('Component', () => {
    test('is instantiated', () => {
        expect(wrapper.isVueInstance).toBeTruthy();
    });
});

describe('NonCheckoutView.vue', () => {
    test('components existence', () => {
        expect(wrapper.findComponent({ name: 'Navbar' }).exists()).toBeTruthy();
    })

    test('default snapshot', () => {
        expect(wrapper.element).toMatchSnapshot();
    })
})