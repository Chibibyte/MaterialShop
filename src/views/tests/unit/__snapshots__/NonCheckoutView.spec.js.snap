// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NonCheckoutView.vue default snapshot 1`] = `
<div>
  <navbar-stub />
   
  <v-main-stub>
    <router-view-stub />
  </v-main-stub>
</div>
`;
